updated system requirements to remove deprecated OSs

jennyowen opened the pull request.

renetapopova reviewed (approved):
Great! Thank you!

renetapopova reviewed (commented):
@jennyowen, I added the deprecations back. Could you please take a look?

jennyowen commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -103,19 +108,15 @@ For cloud environments, and server-based, on-premise environments:
-| *Ubuntu Server 16.04, 18.04, 20.04, 22.04*       | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
-| *Windows Server 2016, 2019, 2022*                | OracleJDK 17, ZuluJDK 17
+| *Red Hat Enterprise Linux Server 8.10, 9.2, 9.4* | Red Hat OpenJDK 17,  Oracle JDK 17, and ZuluJDK 17
+| *Ubuntu Server 16.04, 18.04, 20.04, 22.04, 24.04*| OpenJDK 17, OracleJDK 17, and ZuluJDK 17
```
Deprecated or not, I think we should remove ubuntu 16.04 and 18.04 from this list since they are not even supported by Ubuntu now without a "pro" license [source](https://ubuntu.com/about/release-cycle).
What do you think @b33f ?

b33f replied:
We can't arbitrarily just remove OSes that haven't been announced as deprecated.  This will upset customers and will have some strong feedback.

b33f commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -86,14 +86,19 @@ For personal use and software development:
-| *Ubuntu Desktop 22.04+*                 | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| Operating System                    | Supported JDK
+| *Debian 11, 12*                     | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| *MacOS 11, 12, 13, 14, 15*          | OpenJDK 17, ZuluJDK 17
```
Again, we cannot remove supportive platforms that haven't been deprecated.

renetapopova commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -86,14 +86,19 @@ For personal use and software development:
-| *MacOS 11, 12*                          | OpenJDK 17, ZuluJDK 17
-| *SuSE Enterprise Desktop 15*            | OpenJDK 17, OracleJDK 17
-| *Ubuntu Desktop 22.04+*                 | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| Operating System                    | Supported JDK
```
I think that would work.

renetapopova commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -86,14 +86,19 @@ For personal use and software development:
-| *SuSE Enterprise Desktop 15*            | OpenJDK 17, OracleJDK 17
-| *Ubuntu Desktop 22.04+*                 | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| Operating System                    | Supported JDK
+| *Debian 11, 12*                     | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
```
```suggestion
| *Debian 11, 12*                     | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
```

renetapopova commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -86,14 +86,19 @@ For personal use and software development:
+| Operating System                    | Supported JDK
+| *Debian 11, 12*                     | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| *MacOS 11, 12, 13, 14, 15*          | OpenJDK 17, ZuluJDK 17
+| *SuSE Enterprise Desktop 15*        | OpenJDK 17, OracleJDK 17
```
```suggestion
| *SuSE Enterprise Desktop 15*        | OpenJDK 17/21, OracleJDK 17/21
```

renetapopova commented on modules/ROOT/pages/installation/requirements.adoc:
```diff
@@ -86,14 +86,19 @@ For personal use and software development:
+| *Debian 11, 12*                     | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
+| *MacOS 11, 12, 13, 14, 15*          | OpenJDK 17, ZuluJDK 17
+| *SuSE Enterprise Desktop 15*        | OpenJDK 17, OracleJDK 17
+| *Ubuntu Desktop 22.04, 24.04*       | OpenJDK 17, OracleJDK 17, and ZuluJDK 17
```
```suggestion
| *Ubuntu Desktop 22.04, 24.04*       | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
```

jennyowen commented:
I've added JDK 21 to the system requirements. 

I removed again the old OSs from the supported list, since there is a deprecation notice right below the table. I'm not so sure about that change. Maybe it would be nice to list the deprecated OSs with a footnote next to them, but the footnote is right next to the table? 

like
```
For personal use and software development:

.Software requirements for personal use and software development

[options="header"]
|===
| Operating System               | Supported JDK
| *Debian 11, 12*                | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
| *MacOS 11[1], 12[1], 13, 14, 15*          | OpenJDK 17, ZuluJDK 17/21
| *SuSE Enterprise Desktop 15*   | OpenJDK 17/21, OracleJDK 17/21
| *Ubuntu Desktop 22.04, 24.04*  | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
| *Windows 10[1], 11*                        | OracleJDK 17/21, ZuluJDK 17/21
====
[1] In Neo4j 5.26 LTS, deploying Neo4j on MacOs 11, 12 and Windows 10 is deprecated and will be removed in the next release.


For cloud environments, and server-based, on-premise environments:

.Software requirements for cloud environments, and server-based, on-premise environments
[options="header"]
|===
| Operating System                                 | Supported JDK
| *Amazon Linux 2022 AMI[2], Amazon Linux 2023 AMI*                          | Amazon Corretto 17/21, and OracleJDK 17/21
| *CentOS Stream 8, 9*                             | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
| *Debian 11, 12*                                  | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
| *Red Hat Enterprise Linux Server 8.10, 9.2, 9.4* | Red Hat OpenJDK 17/21,  Oracle JDK 17/21, and ZuluJDK 17/21
| *Ubuntu Server 16.04[2], 18.04[2], 20.04[2], 22.04, 24.04*                     | OpenJDK 17/21, OracleJDK 17/21, and ZuluJDK 17/21
| *Windows Server 2016[2], 2019[2], 2022, 2025*                      | OracleJDK 17/21, ZuluJDK 17/21
|===
[2] In Neo4j 5.26 LTS, deploying Neo4j on Amazon Linux 2022 AMI, Unbuntu Server 16.04, 18.04, 20.04, and Windows Server 2016, 2019 is deprecated and will be removed in the next release.

```

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2052.surge.sh

Updated pages:
- [System requirements](https://neo4j-docs-operations-2052.surge.sh/operations-manual/5/installation/requirements)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2052.surge.sh

Updated pages:
- [System requirements](https://neo4j-docs-operations-2052.surge.sh/operations-manual/5/installation/requirements)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2052.surge.sh

Updated pages:
- [System requirements](https://neo4j-docs-operations-2052.surge.sh/operations-manual/5/installation/requirements)
<!-- Sticky Pull Request Commentdocs-pr-changes -->
